src/core/net/nat64_translator.cpp

```diff
@@ -596,16 +600,33 @@ void Translator::ProtocolCounters::Count4To6Packet(uint8_t aProtocol, uint64_t a
     mTotal.m4To6Bytes += aPacketSize;
 }
 
-State Translator::GetState(void) const
+void Translator::UpdateState(void)
 {
-    State ret = kStateDisabled;
+    State newState;
+
+    if (mEnabled)
+    {
+        if (mIp4Cidr.mLength > 0 && mNat64Prefix.IsValidNat64())
+        {
+            newState = kStateActive;
+        }
+        else
+        {
+            newState = kStateNotRunning;
+        }
+    }
+    else
+    {
+        newState = kStateDisabled;
+    }
 
-    VerifyOrExit(mEnabled);
-    VerifyOrExit(mIp4Cidr.mLength > 0 && mNat64Prefix.IsValidNat64(), ret = kStateNotRunning);
-    ret = kStateActive;
+    VerifyOrExit(mState != newState);
+    mState = newState;
+
+    Get<Notifier>().Signal(kEventNat64TranslatorStateChanged);
```

Comment:
This may be fine as is, but we can consider using `Get<Notifier>().Update(mState, newState,  kEventNat64TranslatorStateChanged)` ?

The `Notifier::Update()`  has the advantage of handling the `SignalIfFirst()` case (which can be important for some edge-cases (ot stack resetting while next layer driver are still running and assuming old states).  I am not sure if it matter for the `kEventNat64TranslatorStateChanged` but may be safer to use `Update()`. Thoughts?

Reply:
Done